dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/Models/Anthropic/AnthropicIOService.cs

```diff
@@ -0,0 +1,229 @@
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Amazon.BedrockRuntime.Model;
+using Amazon.Runtime.Documents;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+
+/// <summary>
+/// Input-output service for Anthropic Claude model.
+/// </summary>
+internal sealed class AnthropicIOService : IBedrockModelIOService
+{
+    /// <summary>
+    /// Builds InvokeModel request Body parameter with structure as required by Anthropic Claude.
+    /// </summary>
+    /// <param name="modelId">The model ID to be used as a request parameter.</param>
+    /// <param name="prompt">The input prompt for text generation.</param>
+    /// <param name="executionSettings">Optional prompt execution settings.</param>
+    /// <returns></returns>
+    object IBedrockModelIOService.GetInvokeModelRequestBody(string modelId, string prompt, PromptExecutionSettings? executionSettings)
+    {
+        var exec = AmazonClaudeExecutionSettings.FromExecutionSettings(executionSettings);
+        var requestBody = new ClaudeRequest.ClaudeTextGenerationRequest()
+        {
+            Prompt = $"\n\nHuman: {prompt}\n\nAssistant:",
+            Temperature = BedrockModelUtilities.GetExtensionDataValue<double?>(executionSettings?.ExtensionData, "temperature") ?? exec.Temperature,
+            MaxTokensToSample = BedrockModelUtilities.GetExtensionDataValue<int?>(executionSettings?.ExtensionData, "max_tokens_to_sample") ?? exec.MaxTokensToSample,
+            StopSequences = BedrockModelUtilities.GetExtensionDataValue<IList<string>?>(executionSettings?.ExtensionData, "stop_sequences") ?? exec.StopSequences,
+            TopP = BedrockModelUtilities.GetExtensionDataValue<double?>(executionSettings?.ExtensionData, "top_p") ?? exec.TopP,
+            TopK = BedrockModelUtilities.GetExtensionDataValue<int?>(executionSettings?.ExtensionData, "top_k") ?? exec.TopK
+        };
+        return requestBody;
+    }
+
+    /// <summary>
+    /// Extracts the test contents from the InvokeModelResponse as returned by the Bedrock API.
+    /// </summary>
+    /// <param name="response">The InvokeModelResponse object provided by the Bedrock InvokeModelAsync output.</param>
+    /// <returns></returns>
+    IReadOnlyList<TextContent> IBedrockModelIOService.GetInvokeResponseBody(InvokeModelResponse response)
+    {
+        using var memoryStream = new MemoryStream();
+        response.Body.CopyToAsync(memoryStream).ConfigureAwait(false).GetAwaiter().GetResult();
+        memoryStream.Position = 0;
+        using var reader = new StreamReader(memoryStream);
+        var responseBody = JsonSerializer.Deserialize<ClaudeResponse>(reader.ReadToEnd());
+        var textContents = new List<TextContent>();
+        if (!string.IsNullOrEmpty(responseBody?.Completion))
+        {
+            textContents.Add(new TextContent(responseBody.Completion));
+        }
+        return textContents;
+    }
```

Comment:
> [!IMPORTANT]
> Apply for all `IOServices`.

You don't need to allocate a new `MemoryStream` to read the response as the response is not being used anymore.

This also avoid the `GetAwaiter().GetResult()` synchronization anti-pattern.

```suggestion
    IReadOnlyList<TextContent> IBedrockModelIOService.GetInvokeResponseBody(InvokeModelResponse response)
    {
        using var reader = new StreamReader(response.Body);
        var responseBody = JsonSerializer.Deserialize<ClaudeResponse>(reader.ReadToEnd());
        List<TextContent>() textContents = [];
        if (!string.IsNullOrEmpty(responseBody?.Completion))
        {
            textContents.Add(new TextContent(responseBody.Completion));
        }
        return textContents;
    }
```